src/apps/reservation-list/list/reservation-list.tsx

```diff
@@ -100,53 +104,55 @@ const ReservationList: FC<ReservationListProps> = ({ pageSize }) => {
         getReservedPhysical(mapFBSReservationToReservationType(data))
       );
     }
-  }, [isSuccess, data]);
-
-  useEffect(() => {
-    refetchFBS();
-    refetchPublizon();
-    refetchUser();
-  }, [modalIds.length, refetchFBS, refetchPublizon, refetchUser]);
+  }, [isSuccessFBS, data]);
 
   const allListsEmpty =
-    readyForPickupReservationsFBS.length === 0 &&
-    readyForPickupReservationsPublizon.length === 0 &&
-    reservedReservationsFBS.length === 0 &&
-    reservedReservationsPublizon.length === 0;
+    readyForPickupReservationsFBS?.length === 0 &&
+    readyForPickupReservationsPublizon?.length === 0 &&
+    reservedReservationsFBS?.length === 0 &&
+    reservedReservationsPublizon?.length === 0 &&
+    !isLoadingFBS &&
+    !isLoadingPublizon;
 
   return (
     <div className="reservation-list-page">
-      <h1 className="text-header-h1 m-32">{t("headerText")}</h1>
+      <h1 className="text-header-h1 m-32">{t("reservationListHeaderText")}</h1>
       {user && <ReservationPauseToggler user={user} />}
-      {!allListsEmpty && (
-        <>
-          <List
-            pageSize={pageSize}
-            header={t("reservationListReadyForPickupTitleText")}
-            emptyListLabel={t("reservationListReadyForPickupEmptyText")}
-            list={sortByOldestPickupDeadline([
-              ...readyForPickupReservationsFBS,
-              ...readyForPickupReservationsPublizon
-            ])}
-          />
-          <List
-            pageSize={pageSize}
-            emptyListLabel={t("reservationListPhysicalReservationsEmptyText")}
-            header={t("reservationListPhysicalReservationsHeaderText")}
-            list={reservedReservationsFBS}
-          />
-          <List
-            pageSize={pageSize}
-            header={t("reservationListDigitalReservationsHeaderText")}
-            emptyListLabel={t("reservationListDigitalReservationsEmptyText")}
-            list={reservedReservationsPublizon}
-          />
-        </>
-      )}
-      {allListsEmpty && (
-        <div className="m-32">
+      {allListsEmpty ? (
+        <div className="list-reservation-container m-32">
           <EmptyList emptyListText={t("reservationListAllEmptyText")} />
         </div>
+      ) : (
+        <>
+          {readyForPickupReservationsFBS !== null &&
```

Comment:
Again I think you could create sub component.

Reply:
Or do you mean the same as in the comment above?